benchmarks/benchmarks/sparse_linalg_lobpcg.py

```diff
@@ -66,45 +22,119 @@ def __init__(self):
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                el, _ = lobpcg(self.Ac, X, self.Bc, M=a, tol=1e-4,
+                               maxiter=40, largest=False)
+            accuracy = max(abs(ee - el) / ee)
+            assert accuracy < tol
+        elif solver == 'eigsh':
+            B = LinearOperator((n, n), matvec=self.Bc, matmat=self.Bc, dtype='float64')
+            A = LinearOperator((n, n), matvec=self.Ac, matmat=self.Ac, dtype='float64')
+            c = cholesky_banded(self.Ab)
+            a_l = LinearOperator((n, n), matvec=a, matmat=a, dtype='float64')
+            ea, _ = eigsh(B, k=m, M=A, Minv=a_l, which='LA', tol=1e-4, maxiter=50,
+                          v0=rng.normal(size=(n, 1)))
+            accuracy = max(abs(ee - np.sort(1./ea)) / ee)
+            assert accuracy < tol
         else:
-            eigh(self.A, self.B, eigvals_only=True, eigvals=(0, m - 1))
+            ed, _ = eigh(self.Aa, self.Ba, subset_by_index=(0, m - 1))
+            accuracy = max(abs(ee - ed) / ee)
+            assert accuracy < tol
 
     def time_sakurai(self, n, solver):
         m = 3
+        ee = self.eigenvalues[:m]
+        tol = 100 * n * n * n* np.finfo(float).eps
+        rng = np.random.default_rng(0)
+        X = rng.normal(size=(n, m))
         if solver == 'lobpcg':
-            X = np.random.rand(n, m)
-            eigs, vecs, resnh = lobpcg(self.A, X, self.B, tol=1e-6, maxiter=500,
-                                       retResidualNormsHistory=1)
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                el, _ = lobpcg(self.A, X, tol=1e-9, maxiter=5000, largest=False)
+            accuracy = max(abs(ee - el) / ee)
+            assert accuracy < tol
+        elif solver == 'eigsh':
+            a_l = LinearOperator((n, n), matvec=self.A, matmat=self.A, dtype='float64')
+            ea, _ = eigsh(a_l, k=m, which='SA', tol=1e-9, maxiter=15000,
+                          v0=rng.normal(size=(n, 1)))
+            accuracy = max(abs(ee - ea) / ee)
+            assert accuracy < tol
         else:
-            eigh(self.A_dense, self.B_dense, eigvals_only=True, eigvals=(0, m - 1))
+            ed, _ = eigh(self.Aa, subset_by_index=(0, m - 1))
+            accuracy = max(abs(ee - ed) / ee)
+            assert accuracy < tol
 
-    # Retain old benchmark results (remove this if changing the benchmark)
-    time_mikota.version = "a1fb679758f7e5cf79d18cc4930afdff999fccc142fe7a4f63e73b39ab1f58bb"
-    time_sakurai.version = "7c38d449924fb71f777bd408072ecc883b8b05e53a6544e97da3887fbc10b235"
+    def time_sakuraii(self, n, solver):
+        def a(x):
+            return cho_solve_banded((c, False), x)
+        m = 3
+        ee = self.eigenvalues[:m]
+        tol = 10 * n * n * n* np.finfo(float).eps
+        rng = np.random.default_rng(0)
+        X =rng.normal(size=(n, m))
```

Comment:
Ensure spaces on both sides for assignments.
```suggestion
        X = rng.normal(size=(n, m))
```